### PR TITLE
Add Conda build/test files for Symbiyosys and Boolector/Z3 solvers

### PR DESCRIPTION
**Pull request by mballance**
This PR adds support for building Symbiyosys on Linux platforms. 
I'm aware of one open issue that I could use some advice with: obtaining a version. I used the Verilator meta.yaml file as a basis for the Symbiyosys one, but the Verilator version mechanism doesn't work well for Symbiyosys. Any advice on typical approaches?

### CODE REVIEW

File: formal/symbiyosys/build.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+# CFLAGS=`echo ${CFLAGS_ORIG} | sed -e 's/-O[0-9]//g'`
+unset CFLAGS
+
+git clone https://github.com/Boolector/boolector.git
+
+cd ${cwd}/boolector
+
+# ilingeling.c contains a variable named 'clone'
+# This conflicts with the clone function, so we rename it
+sed -i -e 's%cd ${LINGELING_DIR}%cd ${LINGELING_DIR}\nsed -i -e "s/\\<clone\\>/_clone/g" ilingeling.c%' contrib/setup-lingeling.sh
```
Comment (mithro) [inline formal/symbiyosys/build.sh]: This should probably just be a patch?

File: formal/symbiyosys/build.sh [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+if test $? -ne 0; then exit 1; fi
+
+CFLAGS=${CFLAGS_ORIG}
+
+#********************************************************************
+#* Build Z3
+#********************************************************************
+cd ${cwd}
+
+git clone https://github.com/Z3Prover/z3.git z3
```
Comment (mithro) [inline formal/symbiyosys/build.sh]: This should be part of the meta.yaml?

File: formal/symbiyosys/build.sh [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+#********************************************************************
+#* Build Boolector
+#********************************************************************
+cd ${cwd}
+
+# CFLAGS=`echo ${CFLAGS_ORIG} | sed -e 's/-O[0-9]//g'`
+unset CFLAGS
+
+git clone https://github.com/Boolector/boolector.git
```
Comment (mithro) [inline formal/symbiyosys/build.sh]: This should be part of the meta.yaml

File: formal/symbiyosys/build.sh (hunk 2 above)
Comment (mithro) [reply formal/symbiyosys/build.sh]: See https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#source-from-multiple-sources

Comment (mballance) [reply formal/symbiyosys/build.sh]: Nice! Somehow, even after searching I didn't manage to find that description of how to use multiple sources.

**Comment by mballance** [comment]
> Is the issue a missing tag?

Most likely. The SymbiYosys repository has no tags, no releases, and no non-master branches. Is there an accepted way to give a version in this case? I guess we could use the Git commit hash. Would that play okay with Conda's expectations?

**Comment by mithro** [comment]
> > Is the issue a missing tag?
> 
> Most likely. The SymbiYosys repository has no tags, no releases, and no non-master branches. Is there an accepted way to give a version in this case? I guess we could use the Git commit hash. Would that play okay with Conda's expectations?

Clone the repository to a seperate location and add a v0.0 tag to the first ever commit. Then use something like https://github.com/apps/pull to keep the repository in sync with the upstream.

**Comment by mithro** [comment]
Friendly ping!

**Comment by mballance** [comment]
Hey Tim, Thanks for the reminder -- getting these issues address sort of got lost in the shuffle. I think the latest commit addresses your feedback. I've changed to specifying multiple sources, dealt with the version issue, and created a separate patch file.

File: formal/symbiyosys/build.sh (hunk 2 above)
Comment (mballance) [reply formal/symbiyosys/build.sh]: Moved this to meta.yaml

File: formal/symbiyosys/build.sh (hunk 3 above)
Comment (mballance) [reply formal/symbiyosys/build.sh]: moved to meta.yaml

**Comment by mithro** [comment]
@mballance You will want to add `formal/symbiyosys` to the `.travis.yml` file too.

**Comment by mballance** [comment]
Good point, @mithro. I think I got the .travis.yaml file right. I also added an entry to the README

**Comment by mithro** [comment]
@mballance Did you forget to push?

**Comment by mballance** [comment]
Oops... Accidentally pushed to the wrong branch. You should have the changes now

**Comment by mithro** [comment]
LGTM.

**Pull request merged**